test/controllers/howDoYouWantToPay.controller.test.ts

```diff
@@ -0,0 +1,167 @@
+import SessionService from 'app/services/session/session.service'
+import FormValidator from 'app/utils/formValidator.util'
+
+import { generateDissolutionGetResponse } from 'test/fixtures/dissolutionApi.fixtures'
+
+describe('HowDoYouWantToPayController', () => {
+
+  let dissolutionService: DissolutionService
+  let paymentService: PaymentService
+  let sessionService: SessionService
+
+  const TOKEN = 'some-token'
+  const COMPANY_NUMBER = 'ABC123'
+  const REDIRECT_CARD_URL = 'http://card-payment-ui-url'
+
+  let dissolution: DissolutionGetResponse
+  let dissolutionSession: DissolutionSession
+  let validator: FormValidator
+
+  beforeEach(() => {
+    dissolutionService = mock(DissolutionService)
+    paymentService = mock(PaymentService)
+    sessionService = mock(SessionService)
+
+    dissolution = generateDissolutionGetResponse()
+    dissolutionSession = generateDissolutionSession(COMPANY_NUMBER)
+    validator = mock(FormValidator)
+
+    when(sessionService.getAccessToken(anything())).thenReturn(TOKEN)
+    when(sessionService.getDissolutionSession(anything())).thenReturn(dissolutionSession)
+    when(paymentService.generatePaymentURL(TOKEN, anything(), anything())).thenResolve(REDIRECT_CARD_URL)
+    when(dissolutionService.getDissolution(TOKEN, dissolutionSession)).thenResolve(dissolution)
+  })
+
+  function initApp(): Application {
+    return createApp(container => {
+      container.rebind(DissolutionService).toConstantValue(instance(dissolutionService))
+      container.rebind(SessionService).toConstantValue(instance(sessionService))
+      container.rebind(PaymentService).toConstantValue(instance(paymentService))
+      container.rebind(FormValidator).toConstantValue(instance(validator))
+    })
+  }
+
+  describe('GET', () => {
+
+    it('should redirect to the search company page if the application has been paid for', async () => {
+      dissolution.application_status = ApplicationStatus.PAID
+
+      const app = initApp()
+
+      await request(app)
+        .get(HOW_DO_YOU_WANT_TO_PAY_URI)
+        .expect(StatusCodes.MOVED_TEMPORARILY)
+        .expect('Location', SEARCH_COMPANY_URI)
+
+      verify(sessionService.setDissolutionSession(anything(), anything())).never()
+    })
+
+    it('should display the how do you want to pay page if the application has not been paid for', async () => {
+      dissolution.application_status = ApplicationStatus.PENDING_PAYMENT
+
+      const app = initApp()
+
+      const res = await request(app)
+        .get(HOW_DO_YOU_WANT_TO_PAY_URI)
+        .expect(StatusCodes.OK)
+
+      const htmlAssertHelper: HtmlAssertHelper = new HtmlAssertHelper(res.text)
+
+      assert.isTrue(htmlAssertHelper.containsText('h1', 'How do you want to pay for the application?'))
+    })
+  })
+
+  describe('POST', () => {
+    it('should display how do you want to pay page with validation error when no payment type is selected', async() => {
+      const form = {
+        paymentType: ''
+      }
```

Comment:
this should be a fixture i.e. 
```
const form: HowDoYouWantToPayFormModel = generateHowDoYouWantToPayForm()
form.paymentType = ''
```

Reply:
TypeScript will complain because it expects a `PaymentType` (not a string). On a previous project we've created an untyped object in tests like this for these scenarios

Reply:
This particular example isn't possible with the generated fixture

Reply:
Ok, as long as these scenarios are the only ones not using a fixture.

I believe in DfT we used casting to get around this issue i.e.
```
form.paymentType = <PaymentType> ''
form.paymentType = '' as PaymentType
```

Reply:
Ill try with casting